<?xml version="1.0" encoding="UTF-8" ?>
<testsuite tests="1" failures="0" name="jda.test.db.javadb.OsmClientTest" time="2.133" errors="1" skipped="0">
  <properties>
    <property name="java.runtime.name" value="Java(TM) SE Runtime Environment"/>
    <property name="java.vm.version" value="13.0.2+8"/>
    <property name="sun.boot.library.path" value="C:\Program Files\Java\jdk-13.0.2\bin"/>
    <property name="maven.multiModuleProjectDirectory" value="C:\jdomainapp\jda"/>
    <property name="java.vm.vendor" value="Oracle Corporation"/>
    <property name="java.vendor.url" value="https://java.oracle.com/"/>
    <property name="path.separator" value=";"/>
    <property name="guice.disable.misplaced.annotation.check" value="true"/>
    <property name="java.vm.name" value="Java HotSpot(TM) 64-Bit Server VM"/>
    <property name="user.country" value="US"/>
    <property name="sun.java.launcher" value="SUN_STANDARD"/>
    <property name="user.script" value=""/>
    <property name="sun.os.patch.level" value=""/>
    <property name="user.dir" value="C:\jdomainapp\jda"/>
    <property name="java.vm.compressedOopsMode" value="32-bit"/>
    <property name="java.vm.specification.name" value="Java Virtual Machine Specification"/>
    <property name="java.runtime.version" value="13.0.2+8"/>
    <property name="os.arch" value="amd64"/>
    <property name="java.io.tmpdir" value="C:\Users\vinhs\AppData\Local\Temp\"/>
    <property name="line.separator" value="
"/>
    <property name="java.vm.specification.vendor" value="Oracle Corporation"/>
    <property name="user.variant" value=""/>
    <property name="os.name" value="Windows 10"/>
    <property name="classworlds.conf" value="C:\app\apache-maven-3.6.3\bin\..\bin\m2.conf"/>
    <property name="sun.jnu.encoding" value="Cp1252"/>
    <property name="java.library.path" value="C:\Program Files\Java\jdk-13.0.2\bin;C:\Windows\Sun\Java\bin;C:\Windows\system32;C:\Windows;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\Java\jdk-13.0.2\bin\;C:\app\apache-maven-3.6.3\bin\;C:\Program Files\nodejs\;C:\app\apache-zookeeper-3.6.3-bin\bin\;;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\PuTTY\;C:\Users\vinhs\AppData\Local\Microsoft\WindowsApps;C:\Users\vinhs\AppData\Roaming\npm;C:\Users\vinhs\AppData\Local\GitHubDesktop\bin;C:\Users\vinhs\AppData\Local\Programs\MiKTeX\miktex\bin\x64\;."/>
    <property name="maven.conf" value="C:\app\apache-maven-3.6.3\bin\../conf"/>
    <property name="jdk.debug" value="release"/>
    <property name="java.class.version" value="57.0"/>
    <property name="java.specification.name" value="Java Platform API Specification"/>
    <property name="sun.management.compiler" value="HotSpot 64-Bit Tiered Compilers"/>
    <property name="os.version" value="10.0"/>
    <property name="library.jansi.path" value="C:\app\apache-maven-3.6.3\bin\..\lib\jansi-native"/>
    <property name="user.home" value="C:\Users\vinhs"/>
    <property name="user.timezone" value="Asia/Bangkok"/>
    <property name="file.encoding" value="Cp1252"/>
    <property name="java.specification.version" value="13"/>
    <property name="user.name" value="vinhs"/>
    <property name="java.class.path" value="C:\app\apache-maven-3.6.3\bin\..\boot\plexus-classworlds-2.6.0.jar"/>
    <property name="java.vm.specification.version" value="13"/>
    <property name="sun.arch.data.model" value="64"/>
    <property name="sun.java.command" value="org.codehaus.plexus.classworlds.launcher.Launcher clean install -U"/>
    <property name="java.home" value="C:\Program Files\Java\jdk-13.0.2"/>
    <property name="user.language" value="en"/>
    <property name="java.specification.vendor" value="Oracle Corporation"/>
    <property name="java.vm.info" value="mixed mode, sharing"/>
    <property name="java.version" value="13.0.2"/>
    <property name="java.vendor" value="Oracle Corporation"/>
    <property name="sun.stderr.encoding" value="cp437"/>
    <property name="maven.home" value="C:\app\apache-maven-3.6.3\bin\.."/>
    <property name="file.separator" value="\"/>
    <property name="java.version.date" value="2020-01-14"/>
    <property name="java.vendor.url.bug" value="https://bugreport.java.com/bugreport/"/>
    <property name="sun.cpu.endian" value="little"/>
    <property name="sun.io.unicode.encoding" value="UnicodeLittle"/>
    <property name="sun.stdout.encoding" value="cp437"/>
    <property name="sun.cpu.isalist" value="amd64"/>
  </properties>
  <testcase classname="jda.test.db.javadb.OsmClientTest" name="test" time="2.133">
    <error message="[NotPossibleException.FAIL_TO_CREATE_OBJECT] Không thể tạo đối tượng lớp: DODM(Configuration(2,Test))" type="jda.modules.common.exceptions.NotPossibleException">jda.modules.common.exceptions.NotPossibleException: [NotPossibleException.FAIL_TO_CREATE_OBJECT] Không thể tạo đối tượng lớp: DODM(Configuration(2,Test))
	at jda.modules.dodm.DODMBasic.getInstance(DODMBasic.java:83)
	at jda.modules.dodm.DODMBasic.getInstance(DODMBasic.java:109)
	at jda.test.db.javadb.OsmClientTest.test(OsmClientTest.java:25)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at jda.modules.dodm.DODMBasic.getInstance(DODMBasic.java:78)
	... 34 more
Caused by: jda.modules.common.exceptions.NotPossibleException: [NotPossibleException.FAIL_TO_CREATE_OBJECT] Không thể tạo đối tượng lớp: DOM(init)
	at jda.modules.dodm.dom.DOMFactory.createDOM(DOMFactory.java:31)
	at jda.modules.dodm.DODMBasic.&lt;init&gt;(DODMBasic.java:57)
	at jda.modules.dodm.DODM.&lt;init&gt;(DODM.java:32)
	... 40 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at jda.modules.dodm.dom.DOMFactory.createDOM(DOMFactory.java:26)
	... 42 more
Caused by: jda.modules.common.exceptions.NotPossibleException: [NotPossibleException.FAIL_TO_CREATE_OBJECT] Không thể tạo đối tượng lớp: JavaDbOSMBasic(OsmConfig (jdbc:derby://localhost/data/DomainAppTest))
	at jda.modules.dodm.osm.OSMFactory.getOsmInstance(OSMFactory.java:51)
	at jda.modules.dodm.dom.DOMBasic.&lt;init&gt;(DOMBasic.java:146)
	at jda.modules.dodm.dom.DOM.&lt;init&gt;(DOM.java:58)
	... 48 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at jda.modules.dodm.osm.OSMFactory.getOsmInstance(OSMFactory.java:46)
	... 50 more
Caused by: jda.modules.common.exceptions.DataSourceException: [DataSourceException.FAIL_TO_CONNECT] Lỗi tạo kết nối đến nguồn dữ liệu: jdbc:derby://localhost/data/DomainAppTest (java.net.ConnectException : Error connecting to server localhost on port 1,527 with message Connection refused: connect.)
	at jda.modules.dodm.osm.relational.RelationalOSMBasic.connect(RelationalOSMBasic.java:215)
	at jda.modules.dodm.osm.relational.RelationalOSMBasic.&lt;init&gt;(RelationalOSMBasic.java:126)
	at jda.modules.dodm.osm.javadb.JavaDbOSMBasic.&lt;init&gt;(JavaDbOSMBasic.java:30)
	... 56 more
Caused by: java.sql.SQLNonTransientConnectionException: java.net.ConnectException : Error connecting to server localhost on port 1,527 with message Connection refused: connect.
	at org.apache.derby.client.am.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:74)
	at org.apache.derby.client.am.SqlException.getSQLException(SqlException.java:325)
	at org.apache.derby.client.ClientAutoloadedDriver.connect(ClientAutoloadedDriver.java:177)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:189)
	at jda.modules.dodm.osm.relational.RelationalOSMBasic.connect(RelationalOSMBasic.java:206)
	... 58 more
Caused by: ERROR 08001: java.net.ConnectException : Error connecting to server localhost on port 1,527 with message Connection refused: connect.
	at org.apache.derby.client.net.NetAgent.&lt;init&gt;(NetAgent.java:153)
	at org.apache.derby.client.net.NetConnection.newAgent_(NetConnection.java:972)
	at org.apache.derby.client.am.ClientConnection.&lt;init&gt;(ClientConnection.java:361)
	at org.apache.derby.client.net.NetConnection.&lt;init&gt;(NetConnection.java:198)
	at org.apache.derby.client.net.ClientJDBCObjectFactoryImpl.newNetConnection(ClientJDBCObjectFactoryImpl.java:278)
	at org.apache.derby.client.ClientAutoloadedDriver.connect(ClientAutoloadedDriver.java:170)
	... 61 more
Caused by: java.net.ConnectException: Connection refused: connect
	at java.base/sun.nio.ch.Net.connect0(Native Method)
	at java.base/sun.nio.ch.Net.connect(Net.java:493)
	at java.base/sun.nio.ch.Net.connect(Net.java:482)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:588)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:339)
	at java.base/java.net.Socket.connect(Socket.java:603)
	at java.base/java.net.Socket.connect(Socket.java:552)
	at java.base/java.net.Socket.&lt;init&gt;(Socket.java:475)
	at java.base/java.net.Socket.&lt;init&gt;(Socket.java:249)
	at java.base/javax.net.DefaultSocketFactory.createSocket(SocketFactory.java:271)
	at org.apache.derby.client.net.OpenSocketAction.run(OpenSocketAction.java:121)
	at org.apache.derby.client.net.OpenSocketAction.run(OpenSocketAction.java:42)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:554)
	at org.apache.derby.client.net.NetAgent.&lt;init&gt;(NetAgent.java:146)
	... 66 more
</error>
    <system-out>Connecting to Java Db server at: derby://localhost/data/DomainAppTest
</system-out>
  </testcase>
</testsuite>